src/array.jl

```diff
@@ -972,3 +963,29 @@ Base.repeat(a::CatArrOrSub{T, N},
 Base.repeat(a::CatArrOrSub{T, N};
             inner = nothing, outer = nothing) where {T, N} =
     CategoricalArray{T, N}(repeat(refs(a), inner=inner, outer=outer), copy(pool(a)))
+
+# JSON3 writing/reading
+StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
+
+StructTypes.construct(::Type{<:CategoricalArray}, array::Vector) =
+    categoricalgeneral(array)
+function categoricalgeneral(array)
+    if eltype(array) >: Nothing
+        categorical(replace(array, nothing=>missing))
+    else
+        categorical(array)
+    end
+end
+
+StructTypes.construct(::Type{<:CategoricalVector{Union{Missing, T}}},
+    array::Vector) where {T, S} = categoricalmissing(T, array)
```

Comment:
Align `array` with `::Type` on the previous line, and add a line break after `=`. Same below.